### ISSUE TITLE
Improve names of generated types

### ISSUE THREAD

**Issue by aKzenT** [Antaris/RazorEngine#334]
Currently razor generates type names looking like CompiledRazorTemplates.Dynamic.RazorEngine_2d2b6c56c6ca4e50a57f48d29fce95a9

Since we are using multiple templates it is sometimes hard to find out which template caused an exception or log message, as the type name doesn't say anything about the template used to create it.

My suggestion would be to have the names be like:
CompiledRazorTemplates.Dynamic.MyTemplate_cshtml_2d2b6c56c6ca4e50a57f48d29fce95a9

This would help especially in production when other features like debugging are normally not availbable. It should be easy to generate this name for cases where the user provides a filename as a template source (standard case). For other cases it might also be usefull to have some kind of factory method which can generate names or name prefixes for a specific template.

I successfully extended an earlier RazorEngine version with this feature, but had to implement my own CSharpCompilerService with large duplications of code from the original, which is a pain every time we want to update to a new version. At the very least it should be possible to add this functionality without having to duplicate lots of code.

Would you accept a pull request for such a feature? And if yes, how would you like to see this implemented?

**Comment by matthid** [comment]
Yes I would accept a PR, but please leave the "CompiledRazorTemplates.Dynamic.RazorEngine_" prefix as it is used by some projects.

**Comment by aKzenT** [comment]
Do you see this as a "ClassNameGenerator" property on the ITemplateServiceConfiguration or is this too specific to expose this there? Thanks!

**Comment by matthid** [comment]
No that would be fine... I've planned to cleanup the configuration but I'm not sure how, so until this is available such changes are fine.

**Comment by danieljsummers** [comment]
I browsed around the configuration; there are certainly several places impacted if ITemplateServiceConfiguration gets a new property, but they seem isolated.

Regarding the actual isolated change - would [here](https://github.com/Antaris/RazorEngine/blob/master/src/source/RazorEngine.Core/Compilation/CompilerServiceBase.cs#L180) be where the assembly name is generated?  (IOW - is the assembly name generated from the temp directory name?)
